frontend/src/app/containers/ExperimentPage/index.js

```diff
@@ -129,114 +125,122 @@ export class ExperimentDetail extends React.Component {
     }
   }
 
-  renderLocaleWarning() {
-    const { experiment, locale, hasAddon } = this.props;
-    if (hasAddon !== null && locale && ((experiment.locales && !experiment.locales.includes(locale)) || (experiment.locale_blocklist && experiment.locale_blocklist.includes(locale)))) {
-      return (
-        <Warning titleL10nId="localeNotTranslatedWarningTitle"
-                 titleL10nArgs={ JSON.stringify({ locale_code: locale }) }
-                 title="This experiment has not been translated to your language (en)."
-                 subtitleL10nId="localeWarningSubtitle"
-                 subtitle="You can still enable it if you like." />
-      );
-    }
-    return null;
-  }
-
-  renderEolBlock() {
-    const { experiment, isAfterCompletedDate } = this.props;
-    if (!experiment.completed || isAfterCompletedDate(experiment)) { return null; }
-
-    const completedDate = formatDate(experiment.completed);
-    const title = `${experiment.title} is ending on ${completedDate}`;
-
-    return (
-      <Warning titleL10nId="eolIntroMessage"
-               titleL10nArgs={ JSON.stringify({ title: experiment.title, completedDate }) }
-               title={ title }>
-        <Localized id={this.l10nId('eolWarning')}>
-          <div>
-            {parser(experiment.eol_warning)}
-          </div>
-        </Localized>
-        <div className="small-spacer" />
-        <Localized id="eolNoticeLink">
-          <a href="/about" target="_blank" rel="noopener noreferrer">
-            Learn more
-          </a>
-        </Localized>
-      </Warning>
-    );
-  }
-
   render() {
-    const { experiment, experiments, installed, isAfterCompletedDate, isDev,
-            hasAddon, setExperimentLastSeen, clientUUID, installedAddons,
-            setPageTitleL10N } = this.props;
+    const {
```

Comment:
This `index.js` diff is largely useless to read. Main changes are scooping out as much as I could into sub-components and getting very explicit about the props & state in destructuring.